server/pbench/bin/index-pbench

```diff
@@ -321,6 +368,115 @@ class ResultData(object):
+        # latency results have three entries: parameters.benchmark, latency.usec, throughput.trans_sec.
+        # throughput results have two entries: parameters.benchmark, throughput.Gb_sec.
+        # All entries are lists, but paremeters.benchmark is a one-element list, whereas the other lists
+        # may have multiple elements, one for a cumulative entry ("all" clients/servers) and one for each
+        # pair of client/server in the experiment (at least for uperf - fio is TBD).
+
+        if 'latency' in results:
+            # Latency results include a latency['usec'] list (or
+            # latency['clat'|'slat'|???] in the fio case) *and* a
+            # throughput['trans_sec'] list (or throughput['iops_sec']
+            # in the fio case). Each list has elements for "all" and
+            # for each client/server pair (or client?) used in the test.
+            if benchmark == "uperf":
+                latency_sublist = ['usec']
+            elif benchmark == "fio":
+                latency_sublist = ['clat', 'lat', 'slat']
+            else:
+                return
+            for sublist in latency_sublist:
+                latency_list = results['latency'][sublist]
+                for i in range(len(latency_list)):
+                    datum_wrapper = ResultData.make_datum_wrapper(latency_list[i])
+                    tseries = latency_list[i]['timeseries']
+                    for res in tseries:
+                        datum = datum_wrapper
+                        datum.update({"datum_value": res['value']})
+                        yield res['date'], datum
+        if 'throughput' in results:
+            # Throughput results include a throughput.Gb_sec list only
+            if benchmark == "uperf":
+                if 'trans_sec' in results['throughput']:
+                    throughput_sublist = ['trans_sec']
+                elif 'Gb_sec' in results['throughput']:
+                    throughput_sublist = ['Gb_sec']
+                else:
+                    return
+            elif benchmark == "fio":
+                throughput_sublist = ['iops_sec']
+            else:
+                return
+            for sublist in throughput_sublist:
+                throughput_list = results['throughput'][sublist]
+                for i in range(len(throughput_list)):
+                    datum_wrapper = ResultData.make_datum_wrapper(throughput_list[i])
+                    tseries = throughput_list[i]['timeseries']
+                    for res in tseries:
+                        datum = datum_wrapper
+                        datum.update({"datum_value": res['value']})
+                        yield res['date'], datum
+
+        return
+
+    # XXX - copied from ToolData - DRY?
+    def convert_timestamp(self, orig_ts):
+        # Convert the given string timestamp, assumed to be a float in
+        # milliseconds since the epoch, to the expected string timestamp format.
```

Comment:
Can you add to this comment that it is assumed the value is already in the UTC timezone?

Reply:
Yes,